Networking/Networking/ApplicationPassword/RequestConverter.swift

```diff
@@ -0,0 +1,17 @@
+import Alamofire
+
+/// Converter to convert Jetpack tunnel requests into REST API requests if needed
+///
+struct RequestConverter {
+    let credentials: Credentials?
+
+    func convert(_ request: URLRequestConvertible) -> URLRequestConvertible {
+        guard let jetpackRequest = request as? JetpackRequest,
```

Comment:
just wondering, do we only do the conversation from `JetpackRequest`? How about `DotcomRequest` and `WordPressOrgRequest`?

Reply:
1. `DotcomRequest` are sent to WordPress.com servers, and it needs WPCOM authentication token. We cannot talk to WPCOM servers using the application password. 
1. `WordPressOrgRequest` will be sent through `WordPressOrgNetwork` instead of `AlamofireNetwork`. 
     - `WordPressOrgNetwork` solely handles `WordPressOrgRequest`, and it doesn't use `RequestConverter`.

Reply:
TIL, thanks for the explanation. Now I have another question 😅 what are the differences between `WordPressOrgRequest` and `RESTRequest`? from the `asURLRequest` implementation in both structs, the way it sets the URL feels pretty similar 🤔

Reply:
- `WordPressOrgRequest` talks to WordPress in general. https://developer.wordpress.org/rest-api/
- `RESTRequest` talks to WooCommerce plugin. https://woocommerce.com/document/woocommerce-rest-api/

Reply:
> 

    WordPressOrgRequest talks to WordPress in general. https://developer.wordpress.org/rest-api/
    RESTRequest talks to WooCommerce plugin. https://woocommerce.com/document/woocommerce-rest-api/

Just chiming in to clarify: I believe `RESTRequest` can be used to handle requests to all WordPress org and Woo requests. The difference between these two types is the way authentication is handled:
- `WordPressOrgRequest` uses cookie authentication.
- `RESTRequest` uses application password.
We should probably come up with better naming for these and update the documentation for clearer explanations.

Reply:
Thanks for the details, Huong!

`RESTRequest` has this extra parameter/value `let wooApiVersion: WooAPIVersion` when compared to `WordPressOrgRequest`.
I thought the `wooApiVersion` parameter makes it clear that we should use `RESTRequest` for the WooCommerce plugin's REST API-related requests. I looked again now and noticed that `WooAPIVersion` has a case `none` which is an empty string. 

> We should probably come up with better naming for these and update the documentation for clearer explanations.

Maybe we could have a single `struct` instead of two and use the `WooAPIVersion.none` case. What do you think? 

I will create an issue after we finish discussing it here.

Reply:
> Maybe we could have a single struct instead of two and use the WooAPIVersion.none case. What do you think?

This makes sense to me! A few things I notice:
- We're using `WordPressApiValidator` in `WordPressOrgRequest` but this is not necessary. Since `WordPressOrgRequest` is required to conform to `Request`, we can use `PlaceholderDataValidator` instead.
- `WordPressOrgRequest` is not using JSON encoding for `POST` and `PUT` requests like `RESTRequest` does, so unifying them with `RESTRequest` implementation will fix this too.

Let's use `RESTRequest` for both then.

Reply:
Created an issue here https://github.com/woocommerce/woocommerce-ios/issues/8512